From 80a45e04fa63e58a1ffd2b78500c2de84ed0354d Mon Sep 17 00:00:00 2001
From: Matthias Runge <mrunge@redhat.com>
Date: Thu, 26 Mar 2015 14:04:18 +0100
Subject: [PATCH] Replace AnonymousUser with AbstractUser

Django-1.8 added _meta classes for User models,
which aren't supported by AnonymousUsers.

SimpleTest has been deprecated since Django-1.6 and
was now removed.

Partially Implements: blueprint django18

Change-Id: Ie243fd2304421694023f579f49f8fa201e761ba3
---
 openstack_auth/tests/run_tests.py | 10 ++++++++--
 openstack_auth/user.py            |  5 +++--
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/openstack_auth/tests/run_tests.py b/openstack_auth/tests/run_tests.py
index 32ec848..dbb7c8a 100644
--- a/openstack_auth/tests/run_tests.py
+++ b/openstack_auth/tests/run_tests.py
@@ -20,7 +20,10 @@ import sys
 os.environ['DJANGO_SETTINGS_MODULE'] = 'openstack_auth.tests.settings'
 
 import django
-from django.test import simple as test_simple
+if django.VERSION < (1, 8, 0):
+    from django.test import simple as test_simple
+else:
+    from django.test import runner
 
 if hasattr(django, 'setup'):
     django.setup()
@@ -35,7 +38,10 @@ def run(*test_args):
         "..",
     )
     sys.path.insert(0, parent)
-    failures = test_simple.DjangoTestSuiteRunner().run_tests(test_args)
+    if django.VERSION < (1, 8, 0):
+        failures = test_simple.DjangoTestSuiteRunner().run_tests(test_args)
+    else:
+        failures = runner.DiscoverRunner(test_args)
     sys.exit(failures)
 
 
diff --git a/openstack_auth/user.py b/openstack_auth/user.py
index 811fe84..ca0c852 100644
--- a/openstack_auth/user.py
+++ b/openstack_auth/user.py
@@ -116,7 +116,7 @@ class Token(object):
         self.serviceCatalog = auth_ref.service_catalog.get_data()
 
 
-class User(models.AnonymousUser):
+class User(models.AbstractUser):
     """A User class with some extra special sauce for Keystone.
 
     In addition to the standard Django user attributes, this class also has
@@ -190,7 +190,7 @@ class User(models.AnonymousUser):
                  services_region=None, user_domain_id=None,
                  user_domain_name=None, domain_id=None, domain_name=None,
                  project_id=None, project_name=None,
-                 is_federated=False, unscoped_token=None):
+                 is_federated=False, unscoped_token=None, password=None):
         self.id = id
         self.pk = id
         self.token = token
@@ -215,6 +215,7 @@ class User(models.AnonymousUser):
         # Unscoped token is used for listing user's project that works
         # for both federated and keystone user.
         self.unscoped_token = unscoped_token
+        self.password = None
 
         # List of variables to be deprecated.
         self.tenant_id = self.project_id
